bytecomp/bytelink.ml

```diff
@@ -325,7 +325,16 @@ let link_bytecode ?final_name tolink exec_name standalone =
        Bytesections.init_record outchan;
        (* The path to the bytecode interpreter (in use_runtime mode) *)
        if String.length !Clflags.use_runtime > 0 then begin
-         output_string outchan (make_absolute !Clflags.use_runtime);
+         let runtime = make_absolute !Clflags.use_runtime in
+         let runtime =
+           (* shebang mustn't exceed 128 including the #! and \0 *)
+           if String.length runtime > 125 then
+             "/bin/sh\n\
+              exec \"" ^ runtime ^ "\" \"$0\" \"$@\""
```

Comment:
You should use `Filename.quote` here to guard against paths with `"` in them.

Reply:
Better late than never, but this will be addressed in the set of PRs for allowing the compiler to be relocated.

As it happens, it's not a bug at the moment inasmuch as you can't get here because `Symtable.init` doesn't quote the runtime either. Fix will be ready before 4.13.